packages/toolkit/src/view/knowledge/KnowledgeBaseView.tsx

```diff
@@ -0,0 +1,145 @@
+import { GeneralAppPageProp } from "../../lib";
+import { Sidebar } from "./components/Sidebar";
+import { KnowledgeBaseTab } from "./components/KnowledgeBaseTab";
+import { UploadExploreTab } from "./components/UploadExploreTab";
+import { CatalogFilesTab } from "./components/CatalogFilesTab";
+import { MarkdownTab } from "./components/MarkdownTab";
+import * as React from "react";
+import { KnowledgeBase } from "../../lib/vdp-sdk/knowledge/types";
+// import { Button, Icons, LinkButton } from "@instill-ai/design-system";
+import { DELETE_KNOWLEDGE_BASE_TIMEOUT, CREDIT_TIMEOUT } from "./components/undoDeleteTime";
+import { ChunkTab } from "./components/ChunkTab";
+import { ImageTab } from "./components/ImageTab";
+import { Nullable } from "@instill-ai/toolkit";
+import { RetrieveTestTab } from "./components/RetrieveTestTab";
+import CreditUsageNotification from "./components/Notifications/CreditUsageFileNotification";
+
+export type KnowledgeBaseViewProps = GeneralAppPageProp;
+
+export const KnowledgeBaseView = (props: KnowledgeBaseViewProps) => {
+  const [selectedKnowledgeBase, setSelectedKnowledgeBase] =
+    React.useState<KnowledgeBase | null>(null);
+  const [activeTab, setActiveTab] = React.useState("knowledge-base");
+  const [selectedTextOption, setSelectedTextOption] = React.useState(
+    null as Nullable<string>
+  );
+  const [showDeleteMessage, setShowDeleteMessage] = React.useState(false);
+  const [isDeleted, setIsDeleted] = React.useState(false);
+  const [knowledgeBaseToDelete, setKnowledgeBaseToDelete] =
+    React.useState<KnowledgeBase | null>(null);
+  const [showCreditUsage, setShowCreditUsage] = React.useState(false);
+
+
+
+  /* -------------------------------------------------------------------------
+   * Query resource data
+   * -----------------------------------------------------------------------*/
+
+  const handleTabChange = (tab: string) => {
+    setActiveTab(tab);
+  };
+
+  const handleTextOptionChange = (option: string | null) => {
+    setSelectedTextOption(option);
+  };
+
+  const handleKnowledgeBaseSelect = (knowledgeBase: KnowledgeBase) => {
+    setSelectedKnowledgeBase(knowledgeBase);
+    handleTabChange("upload");
+  };
+
+  const handleDeleteKnowledgeBase = (knowledgeBase: KnowledgeBase) => {
+    setKnowledgeBaseToDelete(knowledgeBase);
+    setIsDeleted(true);
+    setShowDeleteMessage(true);
+    setTimeout(() => {
+      if (isDeleted && knowledgeBaseToDelete) {
+        setShowDeleteMessage(false);
+        setKnowledgeBaseToDelete(null);
+        setIsDeleted(false);
+      }
+    }, DELETE_KNOWLEDGE_BASE_TIMEOUT);
+  };
+
+  const handleProcessFile = () => {
+    setActiveTab("catalog");
+    setShowCreditUsage(true);
+    setTimeout(() => {
+      setShowCreditUsage(false);
+    }, CREDIT_TIMEOUT);
```

Comment:
@thewbuk Could you explain the reason why you didn't use the toast notification and use this component with manual timeout here?